src/Raven.Studio/typescript/models/database/index/indexDefinition.ts

```diff
@@ -122,20 +121,39 @@ class indexDefinition {
 
         this.reduce.extend({
             required: {
-                onlyIf: () => this.hasReduce()
-            }
+                onlyIf: () => this.hasReduce() && !this.reduce()
+            },
+            validation: [
+                {
+                    validator: (reduceContent: string) => {
+                        if (reduceContent && reduceContent.trim())
+                            return true; 
+                    },
+                    message: `Reduce function is empty!`
```

Comment:
I think we don't use exclamation marks in validation messages. Am I right?

Reply:
done - removed the exclamation mark.